src/RendererDX11.cpp

```diff
@@ -187,6 +202,100 @@ void RendererDX11::perform_frustum_culling(std::shared_ptr<Material> const& mate
 {
 }
 
+void RendererDX11::set_light_buffer(std::shared_ptr<Drawable> const& drawable) const
+{
+    HRESULT hr;
+    D3D11_MAPPED_SUBRESOURCE mappedResource;
+    ConstantBufferLight light_data;
```

Comment:
We should initialize structs using `= {};`, then I'm pretty sure we won't need the
```cpp
    else
    {
        light_data.directional_light.direction = glm::vec3(0.0f);
        light_data.directional_light.ambient = glm::vec3(0.0f);
        light_data.directional_light.diffuse = glm::vec3(0.0f);
        light_data.directional_light.specular = glm::vec3(0.0f);
    }
```